components/brave_wallet/browser/ethereum_provider_impl.h

```diff
@@ -80,7 +80,8 @@ class EthereumProviderImpl final
   void SignMessage(const std::string& address,
                    const std::string& message,
                    RequestCallback callback,
-                   base::Value id);
+                   base::Value id,
+                   bool is_eth_sign);
```

Comment:
And also don't do this, `SignMessage` is dedicated for `eth_sign` and `SignTypedMessage` for eip712

Reply:
It looks like `SignMessage` is covering both the `eth_sign` and `personal_sign` APIs. In the ideal case, we'd still want to keep `personal_sign` and deprecate  `eth_sign` only.

The code on [1142 to 1156 in the implementation](https://github.com/brave/brave-core/pull/17266/files/e7101bfec929e64a72dfd809dbd5fe1a439e691e..2f5c8d267b7190d75f91ea919cfd3fe0e4b69119#diff-5b97dccf71a954164c80d8d4453cc9b47cf0c58ae4c771a849187dc5bf80f888R1142-R1156) is what gave me that idea

Reply:
^ this. Originally I had planned to use `is_eip712 == false` as the condition for triggering the P3A event, but since EIP-191 includes both `eth_sign` and `personal_sign`, I felt like I had no choice.

@darkdh alternative suggestions are welcome, thanks

Reply:
If we are aiming to deprecate `eth_sign` then this makes sense.